cms/models.py

```diff
@@ -748,6 +746,20 @@ class Meta:
         help_text="The faculty members to display on this page",
     )
 
+    def save(self, clean=True, user=None, log_action=False, **kwargs):
+        """
+        Updates related courseware object title.
+        """
+        super().save(clean=clean, user=user, log_action=log_action, **kwargs)
+
+        courseware_object = None
+        if self.is_course_page:
+            courseware_object = getattr(self, "course")
+        elif self.is_program_page:
+            courseware_object = getattr(self, "program")
+        courseware_object.title = self.title
+        courseware_object.save()
+
```

Comment:
I think this might be better to do in an atomic transaction since this by any means if one of them fails we don't want the other one to be updated and be out of sync.

Reply:
After a discussion, we decided to change the ordering of saving the objects instead doing an atomic transaction.